src/Fixer/Phpdoc/PhpdocArrayTypeFixer.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Phpdoc;
+
+use PhpCsFixer\AbstractPhpdocTypesFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Preg;
+use PhpCsFixer\Tokenizer\Tokens;
+
+final class PhpdocArrayTypeFixer extends AbstractPhpdocTypesFixer
+{
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isTokenKindFound(T_DOC_COMMENT);
+    }
+
+    public function isRisky(): bool
+    {
+        return true;
+    }
+
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'PHPDoc `array<T>` type must be used instead of `T[]`.',
+            [
+                new CodeSample(<<<'PHP'
+                    <?php
+                    /**
+                     * @param int[] $x
+                     * @param string[][] $y
+                     */
+
+                    PHP),
+            ],
+            null,
+            'Risky when `array` key should be present, but is missing.'
```

Comment:
Are you sure this makes this rule risky? `T[]` does not have information about key type, so it's implicit `array-key`, changing it to `array<T>` is not risky because it has the same implicit key type. It may be an issue on higher PHPStan strictness, though.

There is [slight difference](https://phpstan.org/writing-php-code/phpdoc-types#iterables) how PHPStan interprets it, so _if_ we consider it's a risky change, we should rather point this (using `T[]` in union types).

I just wanted to discuss it because the current riskiness explanation is not super clear 🙂.

Reply:
I would say this rule is non-risky, but if we consider PHPStan this might change how the code is analyzed, not sure what we should do here 🤔

Reply:
For me even `phpdoc_list_type` is non-risky, as it only changes the comment.

I'm happy to go either way, @keradus, any recommendation?

Reply:
the comment for code declaration is part of code for me. we have framework using the phpdoc comment to determine the code and run logic on it, eg to do API/DB sanitization/convertion.

pls treat it as code and if you see type is different (eg `Date <> string`), mark is risky - if not (same type , only other notation, eg `bool <> boolean`), no need to.

Reply:
So, I've updated the risky description as per @Wirone suggestion.